shared/src/main/kotlin/com/egm/stellio/shared/util/ApiUtils.kt

```diff
@@ -15,11 +18,11 @@ object ApiUtils {
     }
 }
 
-fun String.parseTimeParameter(errorMsg: String): ZonedDateTime =
+fun String.parseTimeParameterwithEither(errorMsg: String): Either<String, ZonedDateTime> =
```

Comment:
Maybe you can keep the old function name parseTimeParameter

Reply:
OK